### DISCUSSION TITLE
Grid Engineのparallel envirnment: pe_4などの説明について

### DISCUSSION THREAD

**Discussion by manabuishii** [nig-sc/nigsc_homepage2#278] [Q&A] [answered]
お世話になっております。
石井と申します。

一般区画を使わせていただいておりますが、

```
qconf -spl
```
を実行すると色々でてくる中に、以下のようなものがあります。

```
pe_4
pe_8
```

これらは、

`def_slot 4`
`def_slot 8`

を指定したものと同じでという認識であっていますでしょうか？

またこの場合、メモリは、各キューの説明にあるようなコアあたり４GBや8GBなどを使えるという認識であっていますでしょうか？

以下のページなどを参考にしました。
https://sc.ddbj.nig.ac.jp/software/univa_grid_engine/
https://sc.ddbj.nig.ac.jp/general_analysis_division/ga_introduction

**Comment by manabuishii** [comment]
また、`pe_2`のようなものを作っていただくことは可能でしょうか？＞cpuを２つするPE

**Comment by oogasawa** [comment, marked as answer]
def_slotは必ず一台の計算機にリソースを取ろうとします。下図の例だとメモリはCPUコア当たり20GB、合計20GB x 2コア = 40GBが一台の計算機の中に確保されます。
![image](https://user-images.githubusercontent.com/35567475/217481768-b938cbe8-1c37-4239-9372-23eb7f5a0fe8.png)

pe_Nのほうは 例えばpe_4 2と書くと2台の計算機に4コアずつリソースを取ります。メモリの取られ方は下図の通りだと１CPUコア当たり8GB、各計算機に 4コアx8GB = 32GB ずつ、合計 32GB x 2台 = 64GB 取られます。
![image](https://user-images.githubusercontent.com/35567475/217482310-29910f07-909e-4415-b8af-3de379c63077.png)

他にもパラレルジョブのオプションがあるのでこの説明は近日中にホームページに反映させたいと思います。

**Comment by akatsumata** [comment]
石井さま

説明をスパコンホームページに反映いたしました。

以下よりご覧ください。

[https://sc.ddbj.nig.ac.jp/software/grid_engine/parallel_jobs/#パラレルジョブの種類 (詳細)](https://sc.ddbj.nig.ac.jp/software/grid_engine/parallel_jobs/#%E3%83%91%E3%83%A9%E3%83%AC%E3%83%AB%E3%82%B8%E3%83%A7%E3%83%96%E3%81%AE%E7%A8%AE%E9%A1%9E-%E8%A9%B3%E7%B4%B0)
